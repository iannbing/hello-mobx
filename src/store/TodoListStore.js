import { types } from 'mobx-state-tree';

const TodoItem = types
  .model('TodoItem', {
    title: types.string,
    done: false
  })
  .actions(self => ({
    toggle() {
      self.done = !self.done;
    }
  }));

const TodoListStore = types
  .model('TodoListStore', {
    items: types.array(TodoItem)
  })
  .actions(self => ({
    add: title => {
      self.items.push({ title, done: false });
    },
    remove: title => {
      self.items = self.items.filter(item => item.title !== title);
    },
    load: items => {
      self.items = items ? [...self.items, ...items] : self.items;
    },
    toggle: title => {
      self.items.find(item => item.title === title).toggle();
    }
  }))
  .views(self => ({
    getSearchResults: searchTerm =>
      self.items.filter(item => item.title.includes(searchTerm))
  }));

export default TodoListStore;
