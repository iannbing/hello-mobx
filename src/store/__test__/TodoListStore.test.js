import { getSnapshot, applyAction, clone, testActions } from 'mobx-state-tree';

import TodoListStore from '../TodoListStore';

let todoListStore;

describe('TodoListStore', () => {
  beforeEach(() => {
    todoListStore = TodoListStore.create({ items: [] });
  });
  it('should load items', () => {
    todoListStore.load([
      { title: 'foo', done: true },
      { title: 'bar', done: false }
    ]);
    expect(getSnapshot(todoListStore)).toMatchSnapshot();
  });
});
