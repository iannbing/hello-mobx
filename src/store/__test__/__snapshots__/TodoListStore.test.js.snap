// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListStore should load items 1`] = `
Object {
  "items": Array [
    Object {
      "done": true,
      "title": "foo",
    },
    Object {
      "done": false,
      "title": "bar",
    },
  ],
}
`;
